x-pack/plugins/cloud_security_posture/server/plugin.ts

```diff
@@ -132,8 +132,8 @@ export class CspPlugin
           packagePolicy: NewPackagePolicy,
           soClient: SavedObjectsClientContract
         ): Promise<NewPackagePolicy> => {
-          if (isCspPackage(packagePolicy.package?.name)) {
-            return cleanupCredentials(packagePolicy);
+          if (isCspPackage(packagePolicy.package?.name) && isAwsPackage(packagePolicy)) {
+            return cleanupAwsCredentials(packagePolicy);
```

Comment:
this function assumes an aws package policy. renamed the function name to make it explicit.